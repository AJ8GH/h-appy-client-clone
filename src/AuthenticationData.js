import React, { useState, useEffect } from 'react';
import { showMessage, hideMessage } from 'react-native-flash-message';
import fetch from 'node-fetch';
import AuthContext from './AuthContext';

export default sendAuthenticationData = async (emailData, passwordData, navigation) => {
  const response = await fetch('http://localhost:3000/user/login', {
    method: 'POST',
    headers: {
      Accept: 'application/json',
      'Content-Type': 'application/json',
    },
    body: JSON.stringify({
      email: emailData,
      password: passwordData,
    }),
  });
  const json = await response.json();
  validation(json, navigation);
  console.log(json);
};

const validation = (response, navigation) => {
  if (response.user) {
    showMessage({
      message: 'Sign in successful',
      description: `Welcome back to H-Appy, ${response.user.name}!`,
      type: 'success',
    });
    dispatch({ type: 'SIGN_IN', token: response.token });
    navigation.navigate('Menu');
  } else {
    showMessage({
      message: 'Authentication failed',
      description: "Those details don't match our records",
      type: 'error',
    });
  }
};
