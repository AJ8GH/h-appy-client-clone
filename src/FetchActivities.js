import { useEffect, useState } from 'react';
import fetch from 'node-fetch';

export default function FetchActivities(limit) {
  const [apiData, setApiData] = useState([]);

  useEffect(() => {
    fetch(`https://happy-haddocks.herokuapp.com/activities?limit=${limit}`)
      .then((response) => response.json())
      .then((json) => setApiData(json))
      .catch((error) => console.error(error));
  }, []);

  return apiData;
}

export function FetchCategories() {
  const [apiData, setApiData] = useState([]);

  useEffect(() => {
    fetch('https://happy-haddocks.herokuapp.com/categories')
      .then((response) => response.json())
      .then((json) => setApiData(json))
      .catch((error) => console.error(error));
  }, []);

  return apiData;
}
