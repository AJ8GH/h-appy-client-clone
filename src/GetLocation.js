import * as Location from 'expo-location';
import { useEffect, useState } from 'react';

export default function GetLocation() {
  const [location, setLocation] = useState(null);

  useEffect(() => {
    (async () => {
      const { status } = await Location.requestForegroundPermissionsAsync();
      if (status !== 'granted') {
        setErrorMsg('Permission to access location was denied');
        return;
      }

      const location = await Location.getCurrentPositionAsync({});
      setLocation(location);
    })();
  }, []);

  console.log(location);

  return location;
}
