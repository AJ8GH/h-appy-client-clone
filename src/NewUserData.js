import {useState, useEffect } from 'react';
import { showMessage, hideMessage } from "react-native-flash-message";
import fetch from 'node-fetch';

export default sendUserData = async (nameData, usernameData, emailData, passwordData, navigation) => {
      let response = await fetch('http://localhost:3000/user', {
        method: 'POST',
        headers: {
          Accept: 'application/json',
          'Content-Type': 'application/json'
        },
        body: JSON.stringify({
          name: nameData,
          username: usernameData,
          email: emailData,
          password: passwordData,
        })
      });
      let json = await response.json();
      validation(json, navigation);
    };
    
  const validation = (response, navigation) => {
   if (response.user) {
     showMessage({
       message: "Signup successful",
       description: `Welcome to H-Appy, ${response.user.name}!`,
       type: "success",
     });
     navigation.navigate('Menu')
   } else if (response.name && response.name === "MongoError") {
    if ("email" in response.keyPattern) {
      showMessage({
        message: "Email in use",
        description: "That email has already been taken",
        type: "error",
      });
    }
    else if ("username" in response.keyPattern) {
      showMessage({
        message: "Username in use",
        description: "That username has already been taken",
        type: "error",
      });
    }
  }
  else if (response.errors) {
    if (response.errors.email && response.errors.email.name === "ValidatorError") {
      showMessage({
        message: "Email not valid",
        description: "You must submit a valid email",
        type: "error",
      });
    }
    else if (response.errors.password && response.errors.password.name === "ValidatorError") {
      showMessage({
        message: "Password too short",
        description: "Your password must be at least 8 characters long",
        type: "error",
      });
    }
  }
}
 
