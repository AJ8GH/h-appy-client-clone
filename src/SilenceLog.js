import React, { useEffect } from 'react';
import { LogBox } from 'react-native';

export default function SilenceLog() {
  useEffect(() => {
    LogBox.ignoreLogs(['VirtualizedLists should never be nested']);
  }, []);
}
