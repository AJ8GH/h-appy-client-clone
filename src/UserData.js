// eslint-disable-next-line prefer-const
import AsyncStorage from '@react-native-async-storage/async-storage';

let userData = {
  nibbles: [], appetisers: [], mains: [], desserts: [],
};

export const emptyUserData = userData;

const defaultData = {
  nibbles: [
    {
      _id: 'bd7dcbea-c1b1-46c2-aed5-3ad53abb28ba',
      name: 'Go to the Cinema',
      ingredients: [],
    },
    {
      _id: '3ac68afc-c605-48d3-a4f8-fbd91aa97f63',
      name: 'Do a Puzzle',
      ingredients: ['a phone or computer or puzzle book'],
    },
    {
      _id: '58694a0f-3da1-471f-bd96-145571e29d72',
      name: 'Take a long awaited break',
      ingredients: [],
    },
  ],
  appetisers: [
    {
      _id: 'bd7acbea-c1b1-46c2-aed5-3ad53dbb28ba',
      name: 'Do a codewars kata',
      ingredients: ['computer'],
    },
    {
      _id: 'bd0acjea-c4b1-46c2-red5-3ad53abb28ba',
      name: 'Play a piece of music',
      ingredients: ['an instrument', 'somewhere private'],
    },
  ],
  mains: [
    {
      _id: 'ai589cm1-oi5n-alf3-bd96-145571e29d72',
      name: 'Learn a new song on the guitar',
      ingredients: ['a guitar'],
    },
  ],
  desserts: [
    {
      _id: '3ac68afc-dk30-3kf9-a4f8-fbd91aa9d07k',
      name: 'Browse Reddit for 3 hours',
      ingredients: ['a phone', 'Ennui'],
    },
  ],
};

const getData = async (key) => {
  try {
    const data = await AsyncStorage.getItem(key);
    return data != null ? JSON.parse(data) : defaultData[key];
  } catch (e) {
    alert('Failed to retrieve data');
  }
};

export const storeData = async (key, value) => {
  try {
    userData[key].push(value);
    const jsonValue = JSON.stringify(userData[key]);
    await AsyncStorage.setItem(key, jsonValue);
    alert('Activity successfully saved');
  } catch (e) {
    alert('Failed to save the data to the storage');
  }
};

const getAllUserData = async () => {
  const nibbles = await getData('nibbles');
  const appetisers = await getData('appetisers');
  const mains = await getData('mains');
  const desserts = await getData('desserts');
  userData = {
    nibbles,
    appetisers,
    mains,
    desserts,
  };
  return userData;
};

export default getAllUserData;
