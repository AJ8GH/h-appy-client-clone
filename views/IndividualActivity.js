import React, { Component, useEffect, useState } from 'react';
import { DrawerActions, useNavigation } from '@react-navigation/native';
import {
  Platform,
  StyleSheet,
  Text,
  View,
  Button,
  Dimensions,
  TextInput,
} from 'react-native';
import react from 'react';
import Header from '../components/Header';
import Grid from '../components/Grid';

export default function IndividualActivity({ route }) {
  const navigation = useNavigation();

  return (
    <View style={styles.container}>
      <Header />
      <View style={styles.nameFlex}>
        <Text style={styles.name}>{route.params.item.name}</Text>
      </View>

      <View style={styles.descriptionContainer}>
        <Text style={styles.description}>
          Being with animals improves mood and takes away stress. Just half an
          hour with your furry friend per day can change your outlook and enable
          you to be more productive.
        </Text>
      </View>

      <View style={styles.grid}>
        <Grid activity={route.params.item} />
      </View>

      {/* <Button
        style={styles.menuButton}
        title="Back to the Main Menu"
        onPress={() => navigation.navigate('Menu')}
      /> */}
    </View>
  );
}

const styles = StyleSheet.create({
  container: {
    flex: 1,
    backgroundColor: '#fff',
    alignItems: 'center',
    justifyContent: 'center',
    backgroundColor: '#f8f9d4',
    height: Dimensions.get('window').height,
    width: Dimensions.get('window').width,
  },
  description: {
    fontSize: 20,
    lineHeight: 25,
    textAlign: 'center',
  },
  descriptionContainer: {
    width: '80%',
    flex: 0.1,
    marginTop: 100,
    position: 'absolute',
    paddingBottom: 50,
  },
  individualDetail: {
    fontSize: 25,
    padding: 10,
  },
  name: {
    fontSize: 25,
    textAlign: 'center',
    maxWidth: '90%',
    fontFamily: 'Chalkduster',
    color: 'white',
  },
  nameFlex: {
    flex: 0.3,
    top: 100,
    width: '80%',
    minHeight: 160,
    position: 'absolute',
    alignItems: 'center',
    justifyContent: 'center',
    backgroundColor: 'black',
    shadowColor: '#000',
    shadowOffset: {
      width: 0,
      height: 8,
    },
    shadowOpacity: 0.44,
    shadowRadius: 10.32,
    elevation: 16,
  },
  detailsFlex: {
    flex: 0.4,
    marginTop: 40,
    position: 'absolute',
    bottom: 70,
  },
  grid: {
    position: 'absolute',
    bottom: 0,
    flex: 0.3,
  },
  // menuButton: {
  //   position: 'absolute',
  //   bottom: 0,
  //   flex: 0.1
  // },
});
