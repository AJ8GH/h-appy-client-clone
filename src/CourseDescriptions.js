import { back } from 'react-native/Libraries/Animated/src/Easing';

const NIBBLES = [
  'Bite-sized activities for the short of time. ',
  'These morsels are low investment, high-return ways to incrementally ',
  'increase your happiness and productivity, on nibble at a time. ',
  'They can take anywhere from 5 minutes to an hour.',
].join('');

const APPETISERS = [
  'These activities are average only in size. ',
  'When you are ready to commit to something but still want to have the rest of the ',
  'day to enjoy afterwards, this is the course for you. ',
  'Perfect if you only have a morning, afternoon or evening to spare.',
].join('');

const MAINS = [
  'Welcome to the main event. When only serious undertakings will satisfy, ',
  'main activities are what you need.',
  'They can take a full day, or be a goal that you work towards over weeks or months. ',
  'Although they require more in terms of time and energy, the rewards are unmatchable.',
].join('');

const DESSERTS = [
  "Sometimes you need minimal effort and maximum comfort. We've got your back. ",
  'Dessert activities are to be enjoyed as a special treat, because sometimes you just need that.',
  "They can be enjoyed whenever, just don't fill up on them. ",
  'Top tip: These are perfect enjoyed right after completing a main',
].join('');

export {
  NIBBLES, APPETISERS, MAINS, DESSERTS,
};
