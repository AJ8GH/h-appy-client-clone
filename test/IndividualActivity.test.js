import React from 'react';
import renderer from 'react-test-renderer';
import IndividualActivity from '../views/IndividualActivity';
import { useNavigation } from '@react-navigation/native';
import { doesActivityNameExist } from '../src/UserData';

jest.mock('@react-navigation/native');
useNavigation.mockImplementation(() => 'Navigation mocked');

jest.mock('../src/UserData');
doesActivityNameExist.mockImplementation();

describe('IndividualActivity', () => {
  const route = {
    params: {
      item: {
        name: 'test',
        size: 'test',
        description: 'test',
      }
    }
  }
  it('renders correctly', () => {
    const tree = renderer.create(<IndividualActivity route={route}/>).toJSON();
    expect(tree).toMatchSnapshot();
  });
})
