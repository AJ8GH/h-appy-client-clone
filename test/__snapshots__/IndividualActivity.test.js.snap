// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`IndividualActivity renders correctly 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#696773",
      "flex": 1,
      "height": 1334,
      "justifyContent": "center",
      "width": 750,
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#363946",
        "flex": 1,
        "fontSize": 24,
        "fontWeight": "bold",
        "height": 60,
        "justifyContent": "center",
        "maxHeight": 60,
        "position": "absolute",
        "top": 0,
        "width": "100%",
      }
    }
  >
    <Text
      style={
        Object {
          "alignItems": "center",
          "color": "#B1B6A6",
          "fontFamily": "Didot",
          "fontSize": 32,
          "fontWeight": "bold",
          "justifyContent": "center",
          "marginBottom": 20,
        }
      }
    >
      H-Appy
    </Text>
  </View>
  <View
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#363946",
        "borderRadius": 7,
        "elevation": 16,
        "flex": 0.2,
        "justifyContent": "center",
        "minHeight": 120,
        "position": "absolute",
        "shadowColor": "#000",
        "shadowOffset": Object {
          "height": 8,
          "width": 0,
        },
        "shadowOpacity": 0.44,
        "shadowRadius": 10.32,
        "top": 72,
        "width": 697.5,
      }
    }
  >
    <Text
      style={
        Object {
          "color": "white",
          "fontFamily": "Chalkduster",
          "fontSize": 22,
          "maxWidth": "90%",
          "textAlign": "center",
        }
      }
    >
      test
    </Text>
  </View>
  <View
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#B1B6A6",
        "borderRadius": 5,
        "flex": 0.82,
        "justifyContent": "center",
        "marginBottom": "11%",
        "marginTop": "40%",
        "width": 697.5,
      }
    }
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "height": "37%",
          "justifyContent": "center",
          "position": "absolute",
          "top": 10,
          "width": 675,
        }
      }
    >
      <RCTScrollView
        style={
          Object {
            "width": "100%",
          }
        }
      >
        <View>
          <Text
            style={
              Object {
                "color": "#23252E",
                "fontFamily": "Courier",
                "fontSize": 15,
                "lineHeight": 25,
                "padding": 10,
                "position": "absolute",
                "textAlign": "center",
                "top": 5,
              }
            }
          >
            test
          </Text>
        </View>
      </RCTScrollView>
    </View>
    <View
      style={
        Object {
          "bottom": 0,
          "elevation": 16,
          "paddingBottom": 15,
          "position": "absolute",
          "shadowColor": "#000",
          "shadowOffset": Object {
            "height": 8,
            "width": 0,
          },
          "shadowOpacity": 0.44,
          "shadowRadius": 10.32,
        }
      }
    >
      <RCTScrollView
        data={
          Array [
            Object {
              "id": "header0",
              "value": "Size",
            },
            Object {
              "id": "header1",
              "value": "Categories",
            },
            Object {
              "id": "size",
              "value": "test",
            },
            Object {
              "id": "categories",
              "value": "",
            },
            Object {
              "id": "header2",
              "value": "Cost",
            },
            Object {
              "id": "header3",
              "value": "Acccessibility",
            },
            Object {
              "id": "cost",
              "value": "Free :)",
            },
            Object {
              "id": "accessibility",
              "value": "undefined/10",
            },
          ]
        }
        disableVirtualization={false}
        getItem={[Function]}
        getItemCount={[Function]}
        horizontal={false}
        initialNumToRender={10}
        keyExtractor={[Function]}
        maxToRenderPerBatch={10}
        onContentSizeChange={[Function]}
        onEndReachedThreshold={2}
        onLayout={[Function]}
        onMomentumScrollEnd={[Function]}
        onScroll={[Function]}
        onScrollBeginDrag={[Function]}
        onScrollEndDrag={[Function]}
        removeClippedSubviews={false}
        renderItem={[Function]}
        scrollEventThrottle={50}
        stickyHeaderIndices={Array []}
        updateCellsBatchingPeriod={50}
        viewabilityConfigCallbackPairs={Array []}
        windowSize={21}
      >
        <View>
          <View
            onLayout={[Function]}
            style={null}
          >
            <View
              style={
                Object {
                  "flexDirection": "row",
                }
              }
            >
              <View
                style={
                  Object {
                    "height": 86.25,
                    "width": 345,
                  }
                }
              >
                <Text
                  style={
                    Array [
                      Object {
                        "backgroundColor": "#353746",
                        "borderRadius": 4,
                        "color": "#fff",
                        "flex": 1,
                        "fontFamily": "Chalkduster",
                        "fontSize": 18,
                        "marginLeft": 3,
                        "marginRight": 3,
                        "overflow": "hidden",
                        "textAlign": "center",
                      },
                      Object {
                        "lineHeight": 20,
                        "marginBottom": -10,
                        "marginTop": 15,
                        "paddingTop": 5,
                        "textDecorationLine": "underline",
                        "zIndex": 999,
                      },
                    ]
                  }
                >
                  Size
                </Text>
              </View>
              <View
                style={
                  Object {
                    "height": 86.25,
                    "width": 345,
                  }
                }
              >
                <Text
                  style={
                    Array [
                      Object {
                        "backgroundColor": "#353746",
                        "borderRadius": 4,
                        "color": "#fff",
                        "flex": 1,
                        "fontFamily": "Chalkduster",
                        "fontSize": 18,
                        "marginLeft": 3,
                        "marginRight": 3,
                        "overflow": "hidden",
                        "textAlign": "center",
                      },
                      Object {
                        "lineHeight": 20,
                        "marginBottom": -10,
                        "marginTop": 15,
                        "paddingTop": 5,
                        "textDecorationLine": "underline",
                        "zIndex": 999,
                      },
                    ]
                  }
                >
                  Categories
                </Text>
              </View>
            </View>
          </View>
          <View
            onLayout={[Function]}
            style={null}
          >
            <View
              style={
                Object {
                  "flexDirection": "row",
                }
              }
            >
              <View
                style={
                  Object {
                    "height": 172.5,
                    "width": 345,
                  }
                }
              >
                <Text
                  style={
                    Array [
                      Object {
                        "backgroundColor": "#353746",
                        "borderRadius": 4,
                        "color": "#fff",
                        "flex": 1,
                        "fontFamily": "Chalkduster",
                        "fontSize": 18,
                        "marginLeft": 3,
                        "marginRight": 3,
                        "overflow": "hidden",
                        "textAlign": "center",
                      },
                      Object {
                        "paddingTop": "12%",
                      },
                    ]
                  }
                >
                  test
                </Text>
              </View>
              <View
                style={
                  Object {
                    "height": 172.5,
                    "width": 345,
                  }
                }
              >
                <Text
                  style={
                    Array [
                      Object {
                        "backgroundColor": "#353746",
                        "borderRadius": 4,
                        "color": "#fff",
                        "flex": 1,
                        "fontFamily": "Chalkduster",
                        "fontSize": 18,
                        "marginLeft": 3,
                        "marginRight": 3,
                        "overflow": "hidden",
                        "textAlign": "center",
                      },
                      Object {
                        "paddingTop": "12%",
                      },
                    ]
                  }
                >
                  
                </Text>
              </View>
            </View>
          </View>
          <View
            onLayout={[Function]}
            style={null}
          >
            <View
              style={
                Object {
                  "flexDirection": "row",
                }
              }
            >
              <View
                style={
                  Object {
                    "height": 86.25,
                    "width": 345,
                  }
                }
              >
                <Text
                  style={
                    Array [
                      Object {
                        "backgroundColor": "#353746",
                        "borderRadius": 4,
                        "color": "#fff",
                        "flex": 1,
                        "fontFamily": "Chalkduster",
                        "fontSize": 18,
                        "marginLeft": 3,
                        "marginRight": 3,
                        "overflow": "hidden",
                        "textAlign": "center",
                      },
                      Object {
                        "lineHeight": 20,
                        "marginBottom": -10,
                        "marginTop": 15,
                        "paddingTop": 5,
                        "textDecorationLine": "underline",
                        "zIndex": 999,
                      },
                    ]
                  }
                >
                  Cost
                </Text>
              </View>
              <View
                style={
                  Object {
                    "height": 86.25,
                    "width": 345,
                  }
                }
              >
                <Text
                  style={
                    Array [
                      Object {
                        "backgroundColor": "#353746",
                        "borderRadius": 4,
                        "color": "#fff",
                        "flex": 1,
                        "fontFamily": "Chalkduster",
                        "fontSize": 18,
                        "marginLeft": 3,
                        "marginRight": 3,
                        "overflow": "hidden",
                        "textAlign": "center",
                      },
                      Object {
                        "lineHeight": 20,
                        "marginBottom": -10,
                        "marginTop": 15,
                        "paddingTop": 5,
                        "textDecorationLine": "underline",
                        "zIndex": 999,
                      },
                    ]
                  }
                >
                  Acccessibility
                </Text>
              </View>
            </View>
          </View>
          <View
            onLayout={[Function]}
            style={null}
          >
            <View
              style={
                Object {
                  "flexDirection": "row",
                }
              }
            >
              <View
                style={
                  Object {
                    "height": 172.5,
                    "width": 345,
                  }
                }
              >
                <Text
                  style={
                    Array [
                      Object {
                        "backgroundColor": "#353746",
                        "borderRadius": 4,
                        "color": "#fff",
                        "flex": 1,
                        "fontFamily": "Chalkduster",
                        "fontSize": 18,
                        "marginLeft": 3,
                        "marginRight": 3,
                        "overflow": "hidden",
                        "textAlign": "center",
                      },
                      Object {
                        "paddingTop": "12%",
                      },
                    ]
                  }
                >
                  Free :)
                </Text>
              </View>
              <View
                style={
                  Object {
                    "height": 172.5,
                    "width": 345,
                  }
                }
              >
                <Text
                  style={
                    Array [
                      Object {
                        "backgroundColor": "#353746",
                        "borderRadius": 4,
                        "color": "#fff",
                        "flex": 1,
                        "fontFamily": "Chalkduster",
                        "fontSize": 18,
                        "marginLeft": 3,
                        "marginRight": 3,
                        "overflow": "hidden",
                        "textAlign": "center",
                      },
                      Object {
                        "paddingTop": "12%",
                      },
                    ]
                  }
                >
                  undefined/10
                </Text>
              </View>
            </View>
          </View>
        </View>
      </RCTScrollView>
    </View>
  </View>
  <View
    accessible={true}
    focusable={false}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignItems": "center",
        "justifyContent": "center",
        "opacity": 1,
      }
    }
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "#353846",
          "borderRadius": 7,
          "bottom": -6,
          "justifyContent": "center",
          "marginBottom": 5,
          "padding": 10,
          "position": "absolute",
          "width": 697.5,
        }
      }
    >
      <Text
        onPress={[Function]}
        style={
          Object {
            "color": "#B1B6A6",
            "fontFamily": "Courier",
          }
        }
        title="Add to favorites"
      >
        Add to favourites
      </Text>
    </View>
  </View>
  <View
    accessible={true}
    focusable={false}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignItems": "center",
        "justifyContent": "center",
        "opacity": 1,
      }
    }
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "#353846",
          "borderRadius": 7,
          "bottom": -6,
          "justifyContent": "center",
          "padding": 10,
          "width": 697.5,
        }
      }
    >
      <Text
        onPress={[Function]}
        style={
          Object {
            "color": "#B1B6A6",
            "fontFamily": "Courier",
          }
        }
        title="Edit activity and Add to Favourites"
      >
        Edit activity and Add to Favourites
      </Text>
    </View>
  </View>
  <View
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#819595",
        "bottom": 0,
        "flex": 0.2,
        "height": "4%",
        "justifyContent": "center",
        "position": "absolute",
        "width": "100%",
      }
    }
  >
    <View
      accessible={true}
      focusable={false}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
        }
      }
    >
      <Text
        onPress={[Function]}
        style={
          Object {
            "color": "#363946",
            "fontFamily": "Courier",
          }
        }
      >
        Back to the Main Menu
      </Text>
    </View>
  </View>
</View>
`;
